### DISCUSSION TITLE
Is there a way to render HTML tags in the table of contents (toc)?

### DISCUSSION THREAD

**Discussion by Lexachoc** [squidfunk/mkdocs-material#6766] [Q&A] [answered]
For example, I have this in the markdown:
### Silicon dioxide, SiO<sub>2</sub>

but in the table of contents (toc), it is displayed as `Silicon dioxide, SiO2`
however I would like the `<sub>` tag work also in the toc.

I tried to add `| safe`
into this line [      {{ toc_item.title }}](https://github.com/squidfunk/mkdocs-material/blob/25a663fd6a0484294f7340b37137aaf52d0380ec/src/templates/partials/toc-item.html#L27)
```html
<!-- Table of contents item -->
<li class="md-nav__item">
  <a href="{{ toc_item.url }}" class="md-nav__link">
    <span class="md-ellipsis">
      {{ toc_item.title}}
    </span>
  </a>
```
```html
<!-- Table of contents item -->
<li class="md-nav__item">
  <a href="{{ toc_item.url }}" class="md-nav__link">
    <span class="md-ellipsis">
      {{ toc_item.title | safe}}
    </span>
  </a>
```

but it doesn't work.
I'm not sure if it's a bug or if it's intentional that the HTML tags for toc are not taken into account.

Does anyone have any ideas on how to do this? Thank you!

**Comment by squidfunk** [comment, marked as answer]
MkDocs by default does not support it, but if you enable the built-in [typeset](https://squidfunk.github.io/mkdocs-material/plugins/typeset/) plugin, the table of contents and navigation allow for HTML to be rendered. Note that you need to [sponsor the project](https://github.com/sponsors/squidfunk) to get access to the plugin, as it is currently incubating in [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/).

If you have access to Insiders, add the following to `mkdocs.yml` to enable the plugin:

``` yaml
plugins:
  - typeset
```

See it in action in our [plugins documentation](https://squidfunk.github.io/mkdocs-material/plugins/blog/), where we use it for settings.
